packages/data-api/src/validators.ts

```diff
@@ -3,23 +3,19 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { hasNoContent, takesDateForm, takesIdForm, yup, yupUtils } from '@tupaia/utils';
+import { takesDateForm, takesIdForm, yup, yupUtils } from '@tupaia/utils';
 
 export const eventOptionsValidator = yup.object().shape({
   organisationUnitCodes: yup.array().of(yup.string().required()).strict().required(),
   startDate: yup
     .string()
-    .test(
-      yupUtils.yupTestAny([hasNoContent, takesDateForm], 'startDate should be in ISO 8601 format'),
-    ),
+    .test(yupUtils.yupTest(takesDateForm, 'startDate should be in ISO 8601 format')),
```

Comment:
In `yupUtils.yupTestAny`, if one of functions testing passes(`hasNoContent` and `takesDateForm`), it will pass the validation. Even though the input is valid, it will still log the validation error during the validation tests run, especially for `hasNoContent`. The reason is that `hasNoContent` extends `LoggedError` and it will always write the log if it is invalid.

I think it should be fine not to use `hasNoContent` here, because `yup` will handle the same validation well if `.required()` is not attached, plus the startDate and endDate seems to always be defined (see translateOptionsForApi()).

Reply:
Looks like some unit tests are failed. 😀

Reply:
put this on hold.

Reply:
I can't think of any good way to disable the winston log in ValidationError when using yupTestAny and our custom validation functions at the same time. Will revert the changes.